### ISSUE TITLE
Improve slate teleporation range/placement indicators

### ISSUE THREAD

**Issue by nvs** [nvs/gem#273]
Currently, slate teleportation either succeeds or fails. There is nothing to indicate before the attempt whether it will work or not. Some considered options follow.

## Default building/construction pathing

- *Not Suitable*

This method would display green/red squares, indicating whether a 'building' can be placed in the desired location. This is a very user-friendly option. However, research has shown it will not work.

The most obvious way to present this building based pathing is use of the construction menu, which can only be detected with the `EVENT_GAME_BUILD_SUBMENU` event. Determining the player and unit in question is left as an exercise to the map maker. Concerns including avoiding desyncs and attempting to isolate the target unit from a selection group come to mind.

The only other way to get this building based pathing is to use 'Tiny XYZ' abilities. These are intended to be used via items, and as such are difficult to use directly as abilities on the command card. You simply cannot detect their order ID, or an event opening their build menus. Now, using them as intended (as items) would probably suffice. However, that means forcing players to consider items as commands for structures. That is not a desired unintended consequence.

Why is this sort of detection necessary? When using this method, all slates would share the same pathing map. However, in order to allow stacking of unlike kinds, this pathing would have to be disabled on the fly for dissimilar type slates. Being unable to properly detect what slate is displaying the placement pathing essentially makes this method not suitable for our purposes.

## Slate range indicators

- *Suitable*

Basically, circles around each slate to indicate the area around them that a similar type slate cannot be placed. This is the most likely solution. There are two ways to go about this. First, allow players to toggle slates on their own. This manual toggle style may feel cumbersome. The second method would be more automatic and consider the currently select slate. However, issues of determining the currently active slate as part of a selection group could occur.

As for the implementation itself, some sort of dummy unit using channel or something similar to display an indicator similar to attack ground. The goal would be to have it be as accurate as possible.

**Comment by nvs** [comment]
This will be pushed to another milestone. At the moment it is not entirely needed. And the reduction of slate teleportation cooldown/hotkey in #246 makes moving slates easier.

**Comment by nvs** [comment]
Again, this may not be entirely needed. Going to play it by ear, given the upcoming 1.4.0 release.

**Comment by nvs** [comment]
This is not needed. Template slates #277 render this issue pointless.
